_posts/2020-02-06-how-to-configure-a-production-grade-ci-cd-setup-for-apps-and-infrastructure-code.adoc

```diff
@@ -0,0 +1,2045 @@
+  up. In other words, the design and implementation of the pipeline covered in this guide intentionally does not solve
+  the use case of infrastructure code for setting up an environment from scratch. Setting up new environments typically
+  require complex deployment orders and permissions modeling that complicate the task. This makes it hard to automate in
+  a reasonable fashion that still respects the threat model we cover here.
+
+Automated testing and feature toggling strategies for infrastructure code::
+  An important factor of CI/CD pipelines is the existence of automated testing and feature toggles. Automated tests give
+  you the confidence in the code before it is deployed to production. Similarly, feature toggles allow you to partially
+  integrate and deploy code for a feature without enabling it. By doing so, you are able to continuously integrate
+  lareger new developments over time. This guide will briefly introduce automated testing and feature toggles for
+  infrastructure code, but will not do a deep dive on the subject.
+
+
+[[core_concepts]]
+== Core Concepts
+
+[[why_is_it_important_to_have_cicd]]
+=== Why is it important to have CI/CD?
+
+.Components of the International Space Station
+image::/assets/img/guides/infrastructure-cicd-pipeline/iss-components.png[]
+
+To understand the benefits of CI/CD, it is worth exploring the opposite: _late integration and late delivery (LI/LD)_.
+We will explain LI/LD using a thought experiment about building the International Space Station (ISS).
+
+The ISS consists of dozens of components, as shown in the image above. Each component is built by a team from a
+different country, with a central team responsible for managing how to organize the development. In LI/LD,
+you would organize the development by coming up with a design for all the components up front and then have each team go
+off and work on their component in _total isolation_. There is complete trust in the design and the teams such that
+there is no need to check in and integrate the components along the way. When all the teams are done, each team launches
+the component into space and then put all the components together at the same time in space, _for the first time_.
+
+It isn't hard to imagine that this could be disastrous: one team would think the other team was responsible for wiring
+while that team thought everything would be wireless; all the teams would use the metric system except one; everyone cut
+toilets from the scope thinking some other team is sure to include it. Finding all of this out once everything has
+already been built and is floating in outer space means that fixing the problems will be very difficule and expensive.
+
+.Many long living branches merging at the same time have a higher chance of having conflicts.
+image::/assets/img/guides/infrastructure-cicd-pipeline/feature-branch-merge-conflict.png[]
```

Comment:
This is a neat diagram, and illustrates the point well, but all the unlabeled lines and branch concepts might not make sense to the reader yet. Maybe talk more about what the branches are in the caption.

Reply:
Moved this down below the paragraph about comparisons with software development, and updated the caption: c92885e41210ec512b6bada8b0f94f299bf71ca6